src/backend/x11/mod.rs

```diff
@@ -146,72 +126,124 @@ pub enum X11Event {
         window_id: u32,
     },
 
-    /// The last buffer presented to the window has been displayed.
-    ///
-    /// When this event is scheduled, the next frame may be rendered.
-    PresentCompleted {
-        /// XID of the window
-        window_id: u32,
-    },
-
     /// The window has received a request to be closed.
     CloseRequested {
         /// XID of the window
         window_id: u32,
     },
 }
 
-/// Represents an active connection to the X to manage events on the Window provided by the backend.
+/// Represents an active connection to the X.
+///
+/// Can be used to create a [`X11Backend`] or an [`smithay::backend::egl::EGLDisplay`].
 #[derive(Debug)]
-pub struct X11Backend {
-    log: Logger,
-    connection: Arc<RustConnection>,
-    source: X11Source,
-    inner: Arc<Mutex<X11Inner>>,
+pub struct X11Connection {
+    pub(crate) display: *mut xlib_sys::xlib::Display,
+    pub(crate) connection: Arc<XCBConnection>,
+    screen: usize,
+    logger: Logger,
 }
 
-impl X11Backend {
-    /// Initializes the X11 backend by connecting to the X server.
-    pub fn new<L>(logger: L) -> Result<X11Backend, X11Error>
+impl X11Connection {
+    /// Initializes the X11 connection to the X server.
+    pub fn new<L>(logger: L) -> Result<X11Connection, X11Error>
     where
         L: Into<Option<slog::Logger>>,
     {
         let logger = crate::slog_or_fallback(logger).new(o!("smithay_module" => "backend_x11"));
 
         info!(logger, "Connecting to the X server");
 
-        let (connection, screen_number) = RustConnection::connect(None)?;
+        let (connection, display, screen) = unsafe {
+            xlib_sys::xlib::XInitThreads();
+            let display = xlib_sys::xlib::XOpenDisplay(std::ptr::null());
+            if display.is_null() {
+                return Err(X11Error::ConnectionFailed(
+                    x11rb::errors::ConnectError::UnknownError,
+                ));
+            }
+            xlib_sys::xlib_xcb::XSetEventQueueOwner(
+                display,
+                xlib_sys::xlib_xcb::XEventQueueOwner::XCBOwnsEventQueue,
+            );
+            let screen = xlib_sys::xlib::XDefaultScreen(display);
+            let xcb_raw = xlib_sys::xlib_xcb::XGetXCBConnection(display);
+            (
+                XCBConnection::from_raw_xcb_connection(xcb_raw, true)?,
```

Comment:
This should be false https://docs.rs/x11rb/latest/x11rb/xcb_ffi/struct.XCBConnection.html#method.from_raw_xcb_connection

Reason being that Xlib effectively owns the xcb connection, therefore to close this we need to use `XDestroyDisplay`